docs/source/clickhouse.md

```diff
@@ -0,0 +1,134 @@
+│ 222 │ Ella Fitzgerald │
+└─────┴─────────────────┘
+```
+
+ClickHouse also provides functions for visualizing your queries. For example, you can use the [`bar`](https://clickhouse.com/docs/en/sql-reference/functions/other-functions#bar) function to create a bar chart of the danceability of songs:
+
+```bash
+./clickhouse local -q "
+    SELECT
+        round(danceability, 1) AS danceability,
+        bar(count(), 0, max(count()) OVER ()) AS dist
+    FROM url('https://huggingface.co/datasets/maharshipandya/spotify-tracks-dataset/resolve/refs%2Fconvert%2Fparquet/default/train/0000.parquet')
+    GROUP BY danceability
+    ORDER BY danceability ASC
+    SETTINGS enable_url_encoding=0, max_http_get_redirects=1"
+
+┌─danceability─┬─dist─────────────────────────────────────────────────────────────────────────────────┐
+│            0 │ ▍                                                                            	      │
+│      	   0.1 │ ████▎                                                                        	      │
+│      	   0.2 │ █████████████▍                                                               	      │
+│      	   0.3 │ ████████████████████████                                                     	      │
+│      	   0.4 │ ████████████████████████████████████████████▋                                	      │
+│      	   0.5 │ ████████████████████████████████████████████████████████████████████▊        	      │
+│      	   0.6 │ ████████████████████████████████████████████████████████████████████████████████     │
+│      	   0.7 │ ██████████████████████████████████████████████████████████████████████       	      │
+│      	   0.8 │ ██████████████████████████████████████████                                   	      │
+│      	   0.9 │ ██████████▋                                                                  	      │
+│            1 │ ▌                                                                            	      │
+└──────────────┴──────────────────────────────────────────────────────────────────────────────────────┘
+```
+
+To get a deeper understanding about a dataset, ClickHouse provides statistical analysis functions for determining how your data is correlated, calculating statistical hypothesis tests, and more. Take a look at ClickHouse's [List of Aggregate Functions](https://clickhouse.com/docs/en/sql-reference/aggregate-functions/reference) for a complete list of available aggregate functions.
+
+## User-defined function (UDFs)
+
+A user-defined function (UDF) allows you to reuse custom logic. Many Hub datasets are often sharded into more than one Parquet file, so it can be easier and more efficient to create a UDF to list and query all the Parquet files of a given dataset from just the dataset name.
+
+For example, let's create a function to return a list of Parquet files from the [blog_authorship_corpus](https://huggingface.co/datasets/blog_authorship_corpus):
+
+```bash
+./clickhouse local -q "
```

Comment:
problem with using -q is the UDF wont persist across sessions. So this UDF wont be available in the next invocation. You need to actually use a console i.e.

```bash
./clickhouse local
```

and then from the console do all the commands